daemon/container_unix.go

```diff
@@ -1025,11 +1033,11 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 		container.updateSandboxNetworkSettings(sb)
 	}
 
-	if err := ep.Join(sb); err != nil {
+	if err = ep.Join(sb); err != nil {
```

Comment:
this is going to hep with one of the strange stale endpoint issues if in case of failures during join.